api/cloud-resources/v1beta1/azureredisinstance_types.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	featuretypes "github.com/kyma-project/cloud-manager/pkg/feature/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RedisInstanceAzureConfigs struct {
+	// +optional
+	MaxClients string `json:"maxclients,omitempty"`
+	// +optional
+	MaxFragmentationMemoryReserved string `json:"maxfragmentationmemory-reserved,omitempty"`
+	// +optional
+	MaxMemoryDelta string `json:"maxmemory-delta,omitempty"`
+	// +optional
+	MaxMemoryPolicy string `json:"maxmemory-policy,omitempty"`
+	// +optional
+	MaxMemoryReserved string `json:"maxmemory-reserved,omitempty"`
+	// +optional
+	NotifyKeyspaceEvents string `json:"notify-keyspace-events,omitempty"`
+	// +optional
+	ZonalConfiguration string `json:"zonal-configuration,omitempty"`
+}
```

Comment:
Consider using map[string]string over this struct. In GCP and AWS implementation, different Redis versions have different available parameters

Reply:
will be done in next iteration.